retrofit/src/main/java/retrofit2/Retrofit.java

```diff
@@ -156,8 +164,9 @@ public <T> T create(final Class<T> service) {
             new Class<?>[] {service},
             new InvocationHandler() {
               private final Object[] emptyArgs = new Object[0];
-
+              
               @Override
+              @SuppressWarnings("mustcall:return") // Suppressed due to [error: [return] incompatible types in return.] and resource leak checker does not handle this warning. (Discussed with professor already).
```

Comment:
The problem here is the use of reflection, which the checker handles soundly but not precisely.

Effectively, this error is complaining that it is possible that reflectively invoking the method might return something with a must-call obligation (since, in theory, a reflective invocation could do anything). We can suppress this because we know that retrofit isn't reflectively creating resources.